scripts/start-gitlab.sh

```diff
@@ -1,31 +1,42 @@
 #!/bin/bash -e
+
 test "$MAKE_TARGET" == "testacc" || { echo "not starting gitlab!"; exit 0; }
-echo "Starting gitlab container..."
-if [[ -n $GITLAB_LICENSE_FILE ]]
-then
-    extra="-v $PWD/license:/license -e GITLAB_LICENSE_FILE=/license/$GITLAB_LICENSE_FILE"
-    img=gitlab/gitlab-ee
-    if [[ ! -f license/$GITLAB_LICENSE_FILE ]]
-    then
-        echo No license
-        exit 1
-    fi
-else
-    img=gitlab/gitlab-ce
+test -z "$(docker ps -f 'name=gitlab' -q)" || { echo "Gitlab already running"; exit 0; }
+
+_gitlab_image="gitlab/gitlab-ce"
+extra=""
+if test -n "${GITLAB_LICENSE_FILE}"; then
+  _license_dir=${GITLAB_LICENSE_DIR:-.license}
+  _license_file=${GITLAB_LICENSE_FILE:-JulienPivotto.gitlab-license}
+
+  if ! test -f "${_license_dir}/${_license_file}"
+  then
+      echo "No license"
+      exit 1
+  fi
+  _gitlab_image=gitlab/gitlab-ee
+  extra+="-v $PWD/${_license_dir}:/license"
+  extra+=" -e GITLAB_LICENSE_FILE=/license/${_license_file}"
 fi
-docker run -d --rm --name gitlab \
-  -e GITLAB_ROOT_PASSWORD=adminadmin \
-  $extra \
-  -p 127.0.0.1:8080:80 \
-  $img
+
+echo "Starting gitlab container..."
+(
+  set -o xtrace
+  docker run -d --rm \
+    --name gitlab \
+    -e "GITLAB_ROOT_PASSWORD=${GITLAB_ROOT_PASSWORD:-adminadmin}" \
+    -p 127.0.0.1:8080:80 \
+    $extra \
+    "${GITLAB_IMAGE:-$_gitlab_image}"
+)
 
 echo -n "Waiting for gitlab to be ready "
 i=1
 until wget -t 1 127.0.0.1:8080 -O /dev/null -q
 do
     sleep 1
     echo -n .
-    if [[ $((i%3)) == 0 ]]; then echo -n ' '; fi
+    if test $((i%3)) -eq 0; then echo -n ' '; fi
```

Comment:
Why not using the bash internal? we enforce bash on #!

Reply:
Both are considered built-in commands. I switched to `test` to make it consistent with the rest of the if statements. The two constructs are evaluated identically by Bash: https://www.gnu.org/software/bash/manual/bash.html#Bash-Conditional-Expressions

Reply:
TIL :) thanks